src/abstract-interpretation/ainode.ts

```diff
@@ -0,0 +1,48 @@
+import { NodeId, ParentInformation, RNodeWithParent } from '../r-bridge'
+import { assertUnreachable, guard } from '../util/assert'
+import { Domain, unifyDomains } from './domain'
+
+export interface AINode {
+	// The ID of the node that logically holds the domain
+	readonly nodeId:       NodeId
+	// The ID of the whole expression that the domain was calculated from
+	readonly expressionId: NodeId
```

Comment:
is expression the correct reference here? With the separation being arbitrary an example may be beneficial

Reply:
I added an example to each comment. But as we're currently not using the `expressionId` at all, I might just remove it in the future (tbh I can't really remember why I added it in the first place 😅)

Reply:
I think it was to indicated the respective parent expression when operating on the sub-expression level.